add 2 oesign cases

johnvenn opened the pull request:
add 2 new cases for oesign
1. sign the enclave with empty config
2. sign the enclave with config file containing negative values.

oesign-customize-cfg.py was a python script to generate cfg with the above purposes, and also,
in future, add more cases on this test point.

mingweishih commented on tests/tools/oesign-engine/test-enclave/CMakeLists.txt:
```diff
@@ -16,6 +16,14 @@ else()
 
+#Generate an empty file
+add_custom_command(OUTPUT empty.conf
+	COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/oesign-test.py 0)
```
Do you mean using the newly created script instead of the original one?

johnvenn replied:
no, no , this script is just to generate test required config files. We are still using the original script to run test and provide result.

mingweishih replied:
Then, can you point me where is the newly added script being used?

johnvenn replied:
the script is to generate empty.conf and neg.conf, 2 config files for oesign tool test(type of negative test)

mingweishih replied:
There my be some confusion.

My original question is that should the command be
```	
        add_custom_command(OUTPUT empty.conf
	COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/oesign-custmize-cfg.py 0)
```

Instead of

```
       add_custom_command(OUTPUT empty.conf
	      COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/oesign-test.py 0)
```

Otherwise, I didn't see the `oesign-custmize-cfg.py` is used anywhere in this PR.

johnvenn replied:
ok, yes, I forgot to add the dir before this script. I will add it. thank you so much.

mingweishih replied:
you're welcome!

mingweishih commented on tests/tools/oesign-engine/test-enclave/CMakeLists.txt:
```diff
@@ -16,6 +16,14 @@ else()
+
+#Generate a config file with negative value
+add_custom_command(OUTPUT neg.conf
+	COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/oesign-test.py 1)
```
Do you mean using the newly created script instead of the original one?

mingweishih commented:
bors try

mingweishih commented:
> ## try
> Merge conflict.

The conflicting file: `tests/tools/oesign-engine/test-enclave/CMakeLists.txt`

johnvenn commented:
> Thanks for addressing the comments! Looks cleaner now.
> Can you also squash the comments (i.e., into a single commit)?

yes, I have squashed the commits into one commit. thanks a lot for the advice.

mingweishih commented:
> > Thanks for addressing the comments! Looks cleaner now.
> > Can you also squash the comments (i.e., into a single commit)?
> 
> yes, I have squashed the commits into one commit. thanks a lot for the advice.

There still a conflict on the file: `tests/tools/oesign-engine/test-enclave/CMakeLists.txt`.

johnvenn commented:
> > > Thanks for addressing the comments! Looks cleaner now.
> > > Can you also squash the comments (i.e., into a single commit)?
> > 
> > 
> > yes, I have squashed the commits into one commit. thanks a lot for the advice.
> 
> There still a conflict on the file: `tests/tools/oesign-engine/test-enclave/CMakeLists.txt`.

ok, i have resolved the conflicts.

mingweishih commented:
bors try

mingweishih commented:
bors try

anakrish commented on tests/tools/oesign-engine/test-enclave/oesign-custmize-config.py:
```diff
@@ -0,0 +1,53 @@
+import os
+
+def generate_empty_config():
+    print("Generating empyt config")
```
Typo: empyt

johnvenn commented:
some errors happened during merging upstream. close this pr. 

created a new PR https://github.com/openenclave/openenclave/pull/3046/. Thanks for all the advices.
